manifests/config.pp

```diff
@@ -63,13 +63,7 @@
   contain foreman_proxy::module::httpboot
 
   contain foreman_proxy::module::puppet
-  foreman_proxy::provider { [
-      'puppet_proxy_customrun',
-      'puppet_proxy_mcollective',
-      'puppet_proxy_puppet_api',
-      'puppet_proxy_salt',
-      'puppet_proxy_ssh',
-    ]:
+  foreman_proxy::provider { [ 'puppet_proxy_puppet_api', ]:
   }
   foreman_proxy::provider { ['puppet_proxy_legacy', 'puppet_proxy_puppetrun']:
```

Comment:
I think you should add the removed providers to this list (which ensures they aren't present)